src/services/step_generator.rs

```diff
@@ -49,8 +51,29 @@ impl Goal {
         if self.tags.contains(&Tag::Budget) {
             return steps;
         }
-        let start = self.start.unwrap_or(calendar_start);
-        let deadline = self.deadline.unwrap_or(calendar_end);
+        let mut start = self.start.unwrap_or(calendar_start);
+        let mut deadline = self.deadline.unwrap_or(calendar_end);
+
+        if let Some(filter) = self.filters.clone() {
+            let timing_scenario = filter.determine_timing_scenario();
+
+            match timing_scenario {
+                TimingScenario::Overflow => {
+                    dbg!(&start, &deadline);
+                    //- set `start`: subtract a few hours (difference between after and 0:00) from start
+                    let after_time = filter.after_time.unwrap();
+                    let diff = 24 - after_time;
+                    start = start - Duration::hours(diff as i64);
```

Comment:
```suggestion
                    start = start - Duration::hours(24 - filter.after_time.unwrap() as i64);
```
Something like this is still readable - but is one line instead of four - and without a comment.